test/models/lookup_keys/puppetclass_lookup_key_test.rb

```diff
@@ -31,4 +31,41 @@ class PuppetclassLookupKeyTest < ActiveSupport::TestCase
     lookup_key.override = false
     assert lookup_key.valid?
   end
+
+  test "deleting puppetclass should delete smart class parameters" do
+    env1 = FactoryGirl.create(:environment)
```

Comment:
Looks like you can put these in a context 'delete params with class', then move the creation of `env1, puppetclass, lookup_key`, and the `environment_class` that uses `env1` to a `setup` block